lib/rbvmomi/vim.rb

```diff
@@ -61,10 +60,17 @@ def self.connect opts
             vim.serviceContent.sessionManager.Login :userName => opts[:user], :password => opts[:password]
         end
       end
-      unless rev_given
-        rev = vim.serviceContent.about.apiVersion
-        vim.rev = [rev, '6.5'].min
-      end
+      # Version compare
+      vim.rev = [rev, opts[:rev]].min { |a, b|
+        a_parts = a.split('.')
+        b_parts = b.split('.')
+        result = 0
+        [a_parts.size, b_parts.size].min.times {|i|
+          result = (a_parts[i].to_i <=> b_parts[i].to_i)
+          break if result != 0
+        }
+        (result != 0) ? result : (a_parts.size <=> b_parts.size)
+      }
```

Comment:
Just a though, what about using either `Gem::Version` compare from stdlib [ref](https://medium.com/little-programming-joys/how-to-compare-version-strings-in-ruby-ced0916a4b71) or `.to_f` the rev so float `>` will work?

Reply:
I think using `.to_f` would be a good solution if we could guarantee the rev will only ever contain two parts (i.e. `6.5` vs `6.5.1`). I'm not sure we can guarantee that. Using `.to_f` on a number with more than two parts just drops the rest of them. I think using `Gem::Version` may be the right way to go. @evanchaoli It looks like you should just be able to compare the the two versions wrapped in a `Gem::Version` type (i.e. `[Gem::Version.new("6.5"), Gem::Version.new("100.0")].min` or `Gem::Version.new("100.0") > Gem::Version.new("6.5")`).

Reply:
I agree Gem::Version is a better solution. I have pushed a new commit that uses Gem::Version to do the version comparison.